### PR TITLE
Pen drawing mode

### PR DESCRIPTION
**Pull request by AndreasBilke**
This PR is from #261. I squashed all commits into one.

*But*: There are some bugs

- It does not work when using the single screen mode
- Erasing does not work
- Pen Highlight Icon does not change immediately, but when changing the slide

### CODE REVIEW

**Comment by woggle** [comment]
I think I've fixed these issues (and also fixed not merging the code that drew the cursor for the pen correctly) at https://github.com/woggle/pdfpc/tree/drawing2 .

**Comment by AndreasBilke** [comment]
Thank you. Looks good for me. I added @phillipberndt to have an additional review.

**Comment by phillipberndt** [comment]
FYI: Works as expected (and is totally awesome 👍) & code looks good at a first glance, but since this is a lot of code the review will take me some time.

File: src/classes/config_file_reader.vala [hunk 1]
```diff
@@ -82,6 +82,9 @@ namespace pdfpc {
                 bt.keyCode = keycode;
                 bt.modMask = modMask;
                 bt.actionName = fields[2];
+                if (fields.length > 3) {
+                    bt.actionArg = fields[3];
+                }
```
Comment (phillipberndt) [inline src/classes/config_file_reader.vala]: This way, it is not a syntax error to pass an argument to an action that doesn't require one, right?

File: src/classes/window/presenter.vala [hunk 2]
```diff
@@ -278,6 +293,10 @@ namespace pdfpc.Window {
             this.saved_icon = this.load_icon("saved.svg", icon_height);
             this.loaded_icon = this.load_icon("loaded.svg", icon_height);
 
+            this.hilight_icon = this.load_icon("hilight.svg", icon_height);
```
Comment (phillipberndt) [inline src/classes/window/presenter.vala]: Shouldn't this be *highlight*, btw?

File: src/classes/config_file_reader.vala (hunk 1 above)
Comment (AndreasBilke) [inline src/classes/config_file_reader.vala]: I added a new commit.

**Comment by woggle** [comment]
bindMouse probably should also call bt.setActionArg().

**Comment by AndreasBilke** [comment]
Oh, with `GDK_SCALE` we have a problem again.

**Comment by AndreasBilke** [comment]
@woggle Is bindMouse using the actionArg? I can't see it.

**Comment by woggle** [comment]
The bindMouse I wrote didn't allow for arguments at all, but it probably should behave like bind so it's possible for mouse buttons to change pen colors.

**Comment by AndreasBilke** [comment]
@woggle but currently, it doesn't?

**Comment by AndreasBilke** [comment]
The `GDK_SCALE` problem was my fault. I forgot to rebase this branch to the latest master (where this was fixed)

**Comment by woggle** [comment]
Yes, currently bindMouse just rejects those config file lines as having too many arguments

**Comment by AndreasBilke** [comment]
If @phillipberndt gives his okay, I would like to merge it :)

File: icons/README.icons [hunk 3]
```diff
@@ -1,4 +1,6 @@
-All icons by jensbob (https://github.com/jensbob/pdfpc)
+blank.svg, pause.svg, saved.svg, snow.svg icons by jensbob (https://github.com/jensbob/pdfpc)
+
+pen.svg, hilight.svg, eraser.svg icons by Charles Reiss
```
Comment (phillipberndt) [inline icons/README.icons]: highlight

**Pull request merged**